### ISSUE TITLE
Error 'Must use npm@>=5' while using npm 5.4.2

### ISSUE THREAD

**Issue by ixje** [ds300/patch-package#25]
> Eriks-Air:neonwallet erik$ ./node_modules/.bin/patch-package react-navigation
> 
> **ERROR** No package-lock.json, npm-shrinkwrap.json, or yarn.lock file.
> 
> You must use either npm@>=5, yarn, or npm-shrinkwrap to manage this project's
> dependencies.
> 
> Eriks-Air:neonwallet erik$ npm -v
> 5.4.2

Also likely related:

> Eriks-Air:neonwallet erik$ npm -v
> 5.4.2
> Eriks-Air:neonwallet erik$ grep "prepare" package.json
>     "prepare": "patch-package"
> Eriks-Air:neonwallet erik$ npm prepare -- react-navigation
> 
> Usage: npm <command>
> 
> where <command> is one of:
>     access, adduser, bin, bugs, c, cache, completion, config,
>     ddp, dedupe, deprecate, dist-tag, docs, doctor, edit,
>     explore, get, help, help-search, i, init, install,
>     install-test, it, link, list, ln, login, logout, ls,
>     outdated, owner, pack, ping, prefix, prune, publish, rb,
>     rebuild, repo, restart, root, run, run-script, s, se,
>     search, set, shrinkwrap, star, stars, start, stop, t, team,
>     test, tst, un, uninstall, unpublish, unstar, up, update, v,
>     version, view, whoami
> 
> npm <command> -h     quick help on <command>
> npm -l           display full usage info
> npm help <term>  search for help on <term>
> npm help npm     involved overview
> 
> Specify configs in the ini-formatted file:
>     /Users/erik/.npmrc
> or on the command line via: npm <command> --key value
> Config info can be viewed via: npm help config
> 
> npm@5.4.2 /usr/local/lib/node_modules/npm
> 

I've followed all the instructions as far as I can see. I also can't really tell why it's failing. Any help/tips are appreciated

**Comment by ds300** [comment]
Thanks for the report!

`npm prepare -- react-navigation` should be `npm run prepare -- react-navigation`

That's my bad, the README is wrong. I'll update it now.

As for the first issue, that's a new one. The error message is a little misleading. It implies that you might be using an old npm but really it just means that it can't find any lock files. Do you definitely have one? i.e. a `package-lock.json`, `npm-shrinkwrap.json`, or `yarn.lock`? If so, then perhaps you could do me a favour:

1. edit `node_modules/patch-package/dist/detectPackageManager.js` and put the following line at the top of the `detectPackageManager` function:

    console.log("app root path:", appRootPath)

Then re-run patch-package and let me know whether the value that is logged is different from the root folder of your project (where the `package.json` is)?

**Comment by ixje** [comment]
Got it working. What happened is that I was on npm 4.3.0 when I tried to use the package. I updated to npm >= 5 and did not touch any packages since then. It turns out that the lock file is not generated until you at least once touch the node_modules or package.json

In their words:
> package-lock.json is automatically generated for any operations where npm modifies either the node_modules tree, or package.json. 

After removing and reinstalling a package it could run and created a patch file :)
